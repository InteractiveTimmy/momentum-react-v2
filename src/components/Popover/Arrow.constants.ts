import { COLORS } from './Popover.constants';

const CLASS_PREFIX = 'md-popover-arrow';

const DEFAULTS = {
  BACKGROUND: 'primary',
  COLOR: COLORS.PRIMARY,
};

const STYLE = {
  svg: `${CLASS_PREFIX}-svg`,
};

export { CLASS_PREFIX, DEFAULTS, STYLE };
