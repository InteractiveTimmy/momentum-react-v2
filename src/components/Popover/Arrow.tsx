import React, { RefObject } from 'react';
import classnames from 'classnames';
import { DEFAULTS, PLACEMENTS, STYLE } from './Arrow.constants';
import './Popover.style.scss';
import type { Props } from './Arrow.types';

/**
 * The Arrow component.
 */
const Arrow = React.forwardRef((props: Props, ref: RefObject<SVGSVGElement>) => {
  const { className, color, id, placement = PLACEMENTS.BOTTOM, style } = props;

  // Point Reference.
  const pr = {
    end: '',
    start: '',
    tipEnd: '',
    tipPeak: '',
    tipStart: '',
  };

  const isVertical =
    placement.startsWith(PLACEMENTS.TOP) || placement.startsWith(PLACEMENTS.BOTTOM);

  switch (placement) {
    case PLACEMENTS.BOTTOM:
    case PLACEMENTS.BOTTOM_START:
    case PLACEMENTS.BOTTOM_END:
      pr.start = '0 12';
      pr.tipStart = '10 -10';
      pr.tipPeak = '2 -2';
      pr.tipEnd = '4 0';
      pr.end = '10 10';
      break;

    case PLACEMENTS.LEFT:
    case PLACEMENTS.LEFT_START:
    case PLACEMENTS.LEFT_END:
      pr.start = '0 0';
      pr.tipStart = '10 10';
      pr.tipPeak = '2 2';
      pr.tipEnd = '0 4';
      pr.end = '-10 10';
      break;

    case PLACEMENTS.RIGHT:
    case PLACEMENTS.RIGHT_START:
    case PLACEMENTS.RIGHT_END:
      pr.start = '12 24';
      pr.tipStart = '-10 -10';
      pr.tipPeak = '-2 -2';
      pr.tipEnd = '0 -4';
      pr.end = '10 -10';
      break;

    case PLACEMENTS.TOP:
    case PLACEMENTS.TOP_START:
    case PLACEMENTS.TOP_END:
      pr.start = '24 0';
      pr.tipStart = '-10 10';
      pr.tipPeak = '-2 2';
      pr.tipEnd = '-4 0';
      pr.end = '-10 -10';
      break;
  }

  const pathData = `m ${pr.start} l ${pr.tipStart} q ${pr.tipPeak} ${pr.tipEnd} l ${pr.end}`;

  const viewBoxHeight = isVertical ? 12 : 24;
  const viewBoxWidth = isVertical ? 24 : 12;
  const viewBox = `0 0 ${viewBoxWidth} ${viewBoxHeight}`;

  return (
    <svg
      id={id}
      ref={ref}
      className={classnames(STYLE.svg, className)}
      style={style}
      xmlns="http://www.w3.org/svg/2000"
      width={viewBoxWidth}
      height={viewBoxHeight}
      viewBox={viewBox}
      data-side={placement}
    >
      <defs>
        <clipPath id={`popover-arrow-cut-stroke-${placement}`}>
          <path d={pathData} />
        </clipPath>
      </defs>

      <path
        data-color={color || DEFAULTS.COLOR}
        clipPath={`url(#popover-arrow-cut-stroke-${placement})`}
        d={pathData}
      />
    </svg>
  );
});

Arrow.displayName = 'Arrow';
export default Arrow;
