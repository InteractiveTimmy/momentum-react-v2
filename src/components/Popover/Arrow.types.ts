import type { Color, PlacementType } from './Popover.types';
import type { CSSProperties } from 'react';

export interface Props {
  /**
   * Custom class for overriding this component's CSS.
   */
  className?: string;

  /**
   * The color of this ModalArrow.
   */
  color?: Color;

  /**
   * Custom id for overriding this component's CSS.
   */
  id?: string;

  /**
   * The placement of the arrow.
   */
  placement: PlacementType;

  /**
   * Custom style for overriding this component's CSS.
   */
  style?: CSSProperties;
}
