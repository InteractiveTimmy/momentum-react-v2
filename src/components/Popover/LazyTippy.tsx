import React, { FC } from 'react';

import Tippy, { TippyProps } from '@tippyjs/react';

export type LazyTippyProps = TippyProps;

export const LazyTippy: FC<LazyTippyProps> = React.forwardRef((props: LazyTippyProps, ref) => {
  const [mounted, setMounted] = React.useState(false);

  const lazyPlugin = {
    fn: () => ({
      onMount: () => setMounted(true),
      onHidden: () => setMounted(false),
    }),
  };

  const computedProps = { ...props };

  computedProps.plugins = [lazyPlugin, ...(props.plugins || [])];

  if (props.render) {
    computedProps.render = (...args) => (mounted ? props.render(...args) : '');
  } else {
    computedProps.content = mounted ? props.content : '';
  }

  return <Tippy ref={ref} {...computedProps} />;
});

LazyTippy.displayName = 'LazyTippy';
