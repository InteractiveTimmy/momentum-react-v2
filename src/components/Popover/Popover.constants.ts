const CLASS_PREFIX = 'md-popover';

const DEFAULTS = {};

const STYLE = {
  wrapper: `${CLASS_PREFIX}-wrapper`,
};

export { CLASS_PREFIX, DEFAULTS, STYLE };
