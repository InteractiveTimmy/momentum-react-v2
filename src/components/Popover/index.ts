import { default as Popover } from './Popover';
import * as CONSTANTS from './Popover.constants';
import { Props } from './Popover.types';

export { CONSTANTS as POPOVER_CONSTANTS };

export type PopoverProps = Props;

export default Popover;
