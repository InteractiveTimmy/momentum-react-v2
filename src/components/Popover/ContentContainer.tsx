import React, { FC } from 'react';
import classnames from 'classnames';
import Arrow from './Arrow';
import { DEFAULTS, STYLE } from './Popover.constants';
import type { Props } from './Popover.types';
import type { Placement } from './Arrow.types';

interface ContentContainerProps
  extends Pick<Props, 'containerProps' | 'color' | 'className' | 'children' | 'id' | 'style'> {
  /**
   * attrs - Tippy attributes passed in
   */
  attrs: {
    'data-placement': Placement;
    'data-reference-hidden'?: string;
    'data-escaped'?: string;
  };
}

/**
 * The ContentContainer component will wrap the content of the popover (=children)
 * in a styled div, which will ensure padding and a proper styling is applied to the content
 * of the Popover. The ContentContainer also adds the arrow of the popover to the DOM.
 */
const ContentContainer: FC<ContentContainerProps> = (props: ContentContainerProps) => {
  const { attrs, containerProps, color, className, children, id, style } = props;

  return (
    <div
      id={id}
      data-padded={containerProps.isPadded || DEFAULTS.IS_PADDED}
      data-round={containerProps.round || DEFAULTS.ROUND}
      data-color={color || DEFAULTS.COLOR}
      style={style}
      className={classnames(STYLE.wrapper, className)}
      {...attrs}
    >
      {children}
      {/*arrow has to be wrapped in HTML element to allow Tippy to style it*/}
      <div id="arrow" className={classnames(STYLE.arrowWrapper)}>
        <Arrow placement={attrs['data-placement']} color={color} />
      </div>
    </div>
  );
};

export default ContentContainer;
