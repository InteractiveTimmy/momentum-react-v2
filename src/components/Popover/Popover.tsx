import React, { FC } from 'react';
import Tippy from '@tippyjs/react';
import ContentContainer from './ContentContainer';
import './Popover.style.scss';
import type { Props } from './Popover.types';

/**
 * The Popover component allows adding a Popover to whatever provided
 * `triggerComponent`. It will show the Popover after a specific event, which is
 * defined by the provided `trigger` prop.
 *
 * Popover uses @tippyjs/react under the hood - possible attributes for future modification
 * can be found here: https://atomiks.github.io/tippyjs/v6/all-props/
 */
const Popover: FC<Props> = (props: Props) => {
  const {
    visible,
    children,
    trigger = 'click',
    triggerComponent,
    containerProps,
    placement = 'bottom',
    interactive,
    color,
    className,
    id,
    style,
  } = props;

  return (
    <Tippy
      render={(attrs) => (
        <ContentContainer
          attrs={attrs}
          id={id}
          style={style}
          containerProps={containerProps}
          color={color}
          className={className}
        >
          {children}
        </ContentContainer>
      )}
      visible={visible}
      placement={placement}
      trigger={trigger}
      interactive={interactive}
      appendTo="parent"
      popperOptions={{
        modifiers: [
          {
            name: 'arrow',
            options: {
              element: '#arrow', // css selector to point to arrow div
              padding: 5,
            },
          },
        ],
      }}
      // offset + 11px (size of arrow standing out of popover), default offset = 5px
      offset={[0, (containerProps?.offset ?? 5) + 11]}
      animation={false}
    >
      {triggerComponent}
    </Tippy>
  );
};

export default Popover;
