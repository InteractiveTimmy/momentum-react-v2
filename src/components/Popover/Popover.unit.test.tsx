import React from 'react';
import { mount } from 'enzyme';

import Popover, { POPOVER_CONSTANTS as CONSTANTS } from './';

describe('<Popover />', () => {
  describe('snapshot', () => {
    it('should match snapshot', () => {
      expect.assertions(1);

      const container = mount(
        <Popover triggerComponent={<button>Click Me!</button>}>
          <p>Content</p>
        </Popover>
      );

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with className', () => {
      expect.assertions(1);

      const className = 'example-class';

      const container = mount(
        <Popover triggerComponent={<button>Click Me!</button>} className={className}>
          <p>Content</p>
        </Popover>
      );

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with id', () => {
      expect.assertions(1);

      const id = 'example-id';

      const container = mount(
        <Popover triggerComponent={<button>Click Me!</button>} id={id}>
          <p>Content</p>
        </Popover>
      );

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with style', () => {
      expect.assertions(1);

      const style = { color: 'pink' };

      const container = mount(
        <Popover triggerComponent={<button>Click Me!</button>} style={style}>
          <p>Content</p>
        </Popover>
      );

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with color', () => {
      expect.assertions(1);

      const container = mount(
        <Popover triggerComponent={<button>Click Me!</button>} color={CONSTANTS.COLORS.TERTIARY}>
          <p>Content</p>
        </Popover>
      );

      expect(container).toMatchSnapshot();
    });
  });

  describe('attributes', () => {
    it('should have its wrapper class', () => {
      expect.assertions(1);

      const element = mount(
        <Popover triggerComponent={<button>Click Me!</button>}>
          <p>Content</p>
        </Popover>
      )
        .find(Popover)
        .getDOMNode();

      expect(element.classList.contains(CONSTANTS.STYLE.wrapper)).toBe(true);
    });

    it('should have provided class when className is provided', () => {
      expect.assertions(1);

      const className = 'example-class';

      const element = mount(
        <Popover triggerComponent={<button>Click Me!</button>} className={className}>
          <p>Content</p>
        </Popover>
      )
        .find(Popover)
        .getDOMNode();

      expect(element.classList.contains(className)).toBe(true);
    });

    it('should have provided id when id is provided', () => {
      expect.assertions(1);

      const id = 'example-id';

      const element = mount(
        <Popover triggerComponent={<button>Click Me!</button>} id={id}>
          <p>Content</p>
        </Popover>
      )
        .find(Popover)
        .getDOMNode();

      expect(element.id).toBe(id);
    });

    it('should have provided style when style is provided', () => {
      expect.assertions(1);

      const style = { color: 'pink' };
      const styleString = 'color: pink;';

      const element = mount(
        <Popover triggerComponent={<button>Click Me!</button>} style={style}>
          <p>Content</p>
        </Popover>
      )
        .find(Popover)
        .getDOMNode();

      expect(element.getAttribute('style')).toBe(styleString);
    });

    /* ...additional attribute tests... */
  });

  describe('actions', () => {
    /* ...action tests... */
  });
});
