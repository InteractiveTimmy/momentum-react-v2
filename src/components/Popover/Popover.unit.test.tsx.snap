// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Popover /> snapshot should match snapshot 1`] = `
<Popover
  triggerComponent={
    <button>
      Click Me!
    </button>
  }
>
  <ForwardRef(TippyWrapper)
    animation={false}
    appendTo="parent"
    offset={
      Array [
        0,
        16,
      ]
    }
    placement="bottom"
    popperOptions={
      Object {
        "modifiers": Array [
          Object {
            "name": "arrow",
            "options": Object {
              "element": "#arrow",
              "padding": 5,
            },
          },
        ],
      }
    }
    render={[Function]}
    trigger="click"
  >
    <Tippy
      animation={false}
      appendTo="parent"
      offset={
        Array [
          0,
          16,
        ]
      }
      placement="bottom"
      popperOptions={
        Object {
          "modifiers": Array [
            Object {
              "name": "arrow",
              "options": Object {
                "element": "#arrow",
                "padding": 5,
              },
            },
          ],
        }
      }
      render={[Function]}
      trigger="click"
    >
      <button>
        Click Me!
      </button>
      <Portal
        containerInfo={
          <div
            data-tippy-root=""
            id="tippy-1"
            style="pointer-events: none; z-index: 9999;"
          >
            <div
              class="md-popover-wrapper"
              data-color="primary"
              data-padded="true"
              data-round="0"
            >
              <p>
                Content
              </p>
              <div
                class="md-popover-arrow-wrapper"
                id="arrow"
              >
                <svg
                  class="md-popover-arrow-svg"
                  data-side="bottom"
                  height="12"
                  viewBox="0 0 24 12"
                  width="24"
                  xmlns="http://www.w3.org/svg/2000"
                >
                  <defs>
                    <clippath
                      id="popover-arrow-cut-stroke-bottom"
                    >
                      <path
                        d="m 0 12 l 10 -10 q 2 -2 4 0 l 10 10"
                      />
                    </clippath>
                  </defs>
                  <path
                    clip-path="url(#popover-arrow-cut-stroke-bottom)"
                    d="m 0 12 l 10 -10 q 2 -2 4 0 l 10 10"
                    data-color="primary"
                  />
                </svg>
              </div>
            </div>
          </div>
        }
      >
        <ContentContainer
          attrs={
            Object {
              "data-placement": undefined,
            }
          }
        >
          <div
            className="md-popover-wrapper"
            data-color="primary"
            data-padded={true}
            data-round={0}
          >
            <p>
              Content
            </p>
            <div
              className="md-popover-arrow-wrapper"
              id="arrow"
            >
              <Arrow>
                <svg
                  className="md-popover-arrow-svg"
                  data-side="bottom"
                  height={12}
                  viewBox="0 0 24 12"
                  width={24}
                  xmlns="http://www.w3.org/svg/2000"
                >
                  <defs>
                    <clipPath
                      id="popover-arrow-cut-stroke-bottom"
                    >
                      <path
                        d="m 0 12 l 10 -10 q 2 -2 4 0 l 10 10"
                      />
                    </clipPath>
                  </defs>
                  <path
                    clipPath="url(#popover-arrow-cut-stroke-bottom)"
                    d="m 0 12 l 10 -10 q 2 -2 4 0 l 10 10"
                    data-color="primary"
                  />
                </svg>
              </Arrow>
            </div>
          </div>
        </ContentContainer>
      </Portal>
    </Tippy>
  </ForwardRef(TippyWrapper)>
</Popover>
`;

exports[`<Popover /> snapshot should match snapshot with className 1`] = `
<Popover
  className="example-class"
  triggerComponent={
    <button>
      Click Me!
    </button>
  }
>
  <ForwardRef(TippyWrapper)
    animation={false}
    appendTo="parent"
    offset={
      Array [
        0,
        16,
      ]
    }
    placement="bottom"
    popperOptions={
      Object {
        "modifiers": Array [
          Object {
            "name": "arrow",
            "options": Object {
              "element": "#arrow",
              "padding": 5,
            },
          },
        ],
      }
    }
    render={[Function]}
    trigger="click"
  >
    <Tippy
      animation={false}
      appendTo="parent"
      offset={
        Array [
          0,
          16,
        ]
      }
      placement="bottom"
      popperOptions={
        Object {
          "modifiers": Array [
            Object {
              "name": "arrow",
              "options": Object {
                "element": "#arrow",
                "padding": 5,
              },
            },
          ],
        }
      }
      render={[Function]}
      trigger="click"
    >
      <button>
        Click Me!
      </button>
      <Portal
        containerInfo={
          <div
            data-tippy-root=""
            id="tippy-2"
            style="pointer-events: none; z-index: 9999;"
          >
            <div
              class="md-popover-wrapper example-class"
              data-color="primary"
              data-padded="true"
              data-round="0"
            >
              <p>
                Content
              </p>
              <div
                class="md-popover-arrow-wrapper"
                id="arrow"
              >
                <svg
                  class="md-popover-arrow-svg"
                  data-side="bottom"
                  height="12"
                  viewBox="0 0 24 12"
                  width="24"
                  xmlns="http://www.w3.org/svg/2000"
                >
                  <defs>
                    <clippath
                      id="popover-arrow-cut-stroke-bottom"
                    >
                      <path
                        d="m 0 12 l 10 -10 q 2 -2 4 0 l 10 10"
                      />
                    </clippath>
                  </defs>
                  <path
                    clip-path="url(#popover-arrow-cut-stroke-bottom)"
                    d="m 0 12 l 10 -10 q 2 -2 4 0 l 10 10"
                    data-color="primary"
                  />
                </svg>
              </div>
            </div>
          </div>
        }
      >
        <ContentContainer
          attrs={
            Object {
              "data-placement": undefined,
            }
          }
          className="example-class"
        >
          <div
            className="md-popover-wrapper example-class"
            data-color="primary"
            data-padded={true}
            data-round={0}
          >
            <p>
              Content
            </p>
            <div
              className="md-popover-arrow-wrapper"
              id="arrow"
            >
              <Arrow>
                <svg
                  className="md-popover-arrow-svg"
                  data-side="bottom"
                  height={12}
                  viewBox="0 0 24 12"
                  width={24}
                  xmlns="http://www.w3.org/svg/2000"
                >
                  <defs>
                    <clipPath
                      id="popover-arrow-cut-stroke-bottom"
                    >
                      <path
                        d="m 0 12 l 10 -10 q 2 -2 4 0 l 10 10"
                      />
                    </clipPath>
                  </defs>
                  <path
                    clipPath="url(#popover-arrow-cut-stroke-bottom)"
                    d="m 0 12 l 10 -10 q 2 -2 4 0 l 10 10"
                    data-color="primary"
                  />
                </svg>
              </Arrow>
            </div>
          </div>
        </ContentContainer>
      </Portal>
    </Tippy>
  </ForwardRef(TippyWrapper)>
</Popover>
`;

exports[`<Popover /> snapshot should match snapshot with color 1`] = `
<Popover
  color="tertiary"
  triggerComponent={
    <button>
      Click Me!
    </button>
  }
>
  <ForwardRef(TippyWrapper)
    animation={false}
    appendTo="parent"
    offset={
      Array [
        0,
        16,
      ]
    }
    placement="bottom"
    popperOptions={
      Object {
        "modifiers": Array [
          Object {
            "name": "arrow",
            "options": Object {
              "element": "#arrow",
              "padding": 5,
            },
          },
        ],
      }
    }
    render={[Function]}
    trigger="click"
  >
    <Tippy
      animation={false}
      appendTo="parent"
      offset={
        Array [
          0,
          16,
        ]
      }
      placement="bottom"
      popperOptions={
        Object {
          "modifiers": Array [
            Object {
              "name": "arrow",
              "options": Object {
                "element": "#arrow",
                "padding": 5,
              },
            },
          ],
        }
      }
      render={[Function]}
      trigger="click"
    >
      <button>
        Click Me!
      </button>
      <Portal
        containerInfo={
          <div
            data-tippy-root=""
            id="tippy-5"
            style="pointer-events: none; z-index: 9999;"
          >
            <div
              class="md-popover-wrapper"
              data-color="tertiary"
              data-padded="true"
              data-round="0"
            >
              <p>
                Content
              </p>
              <div
                class="md-popover-arrow-wrapper"
                id="arrow"
              >
                <svg
                  class="md-popover-arrow-svg"
                  data-side="bottom"
                  height="12"
                  viewBox="0 0 24 12"
                  width="24"
                  xmlns="http://www.w3.org/svg/2000"
                >
                  <defs>
                    <clippath
                      id="popover-arrow-cut-stroke-bottom"
                    >
                      <path
                        d="m 0 12 l 10 -10 q 2 -2 4 0 l 10 10"
                      />
                    </clippath>
                  </defs>
                  <path
                    clip-path="url(#popover-arrow-cut-stroke-bottom)"
                    d="m 0 12 l 10 -10 q 2 -2 4 0 l 10 10"
                    data-color="tertiary"
                  />
                </svg>
              </div>
            </div>
          </div>
        }
      >
        <ContentContainer
          attrs={
            Object {
              "data-placement": undefined,
            }
          }
          color="tertiary"
        >
          <div
            className="md-popover-wrapper"
            data-color="tertiary"
            data-padded={true}
            data-round={0}
          >
            <p>
              Content
            </p>
            <div
              className="md-popover-arrow-wrapper"
              id="arrow"
            >
              <Arrow
                color="tertiary"
              >
                <svg
                  className="md-popover-arrow-svg"
                  data-side="bottom"
                  height={12}
                  viewBox="0 0 24 12"
                  width={24}
                  xmlns="http://www.w3.org/svg/2000"
                >
                  <defs>
                    <clipPath
                      id="popover-arrow-cut-stroke-bottom"
                    >
                      <path
                        d="m 0 12 l 10 -10 q 2 -2 4 0 l 10 10"
                      />
                    </clipPath>
                  </defs>
                  <path
                    clipPath="url(#popover-arrow-cut-stroke-bottom)"
                    d="m 0 12 l 10 -10 q 2 -2 4 0 l 10 10"
                    data-color="tertiary"
                  />
                </svg>
              </Arrow>
            </div>
          </div>
        </ContentContainer>
      </Portal>
    </Tippy>
  </ForwardRef(TippyWrapper)>
</Popover>
`;

exports[`<Popover /> snapshot should match snapshot with id 1`] = `
<Popover
  id="example-id"
  triggerComponent={
    <button>
      Click Me!
    </button>
  }
>
  <ForwardRef(TippyWrapper)
    animation={false}
    appendTo="parent"
    offset={
      Array [
        0,
        16,
      ]
    }
    placement="bottom"
    popperOptions={
      Object {
        "modifiers": Array [
          Object {
            "name": "arrow",
            "options": Object {
              "element": "#arrow",
              "padding": 5,
            },
          },
        ],
      }
    }
    render={[Function]}
    trigger="click"
  >
    <Tippy
      animation={false}
      appendTo="parent"
      offset={
        Array [
          0,
          16,
        ]
      }
      placement="bottom"
      popperOptions={
        Object {
          "modifiers": Array [
            Object {
              "name": "arrow",
              "options": Object {
                "element": "#arrow",
                "padding": 5,
              },
            },
          ],
        }
      }
      render={[Function]}
      trigger="click"
    >
      <button>
        Click Me!
      </button>
      <Portal
        containerInfo={
          <div
            data-tippy-root=""
            id="tippy-3"
            style="pointer-events: none; z-index: 9999;"
          >
            <div
              class="md-popover-wrapper"
              data-color="primary"
              data-padded="true"
              data-round="0"
              id="example-id"
            >
              <p>
                Content
              </p>
              <div
                class="md-popover-arrow-wrapper"
                id="arrow"
              >
                <svg
                  class="md-popover-arrow-svg"
                  data-side="bottom"
                  height="12"
                  viewBox="0 0 24 12"
                  width="24"
                  xmlns="http://www.w3.org/svg/2000"
                >
                  <defs>
                    <clippath
                      id="popover-arrow-cut-stroke-bottom"
                    >
                      <path
                        d="m 0 12 l 10 -10 q 2 -2 4 0 l 10 10"
                      />
                    </clippath>
                  </defs>
                  <path
                    clip-path="url(#popover-arrow-cut-stroke-bottom)"
                    d="m 0 12 l 10 -10 q 2 -2 4 0 l 10 10"
                    data-color="primary"
                  />
                </svg>
              </div>
            </div>
          </div>
        }
      >
        <ContentContainer
          attrs={
            Object {
              "data-placement": undefined,
            }
          }
          id="example-id"
        >
          <div
            className="md-popover-wrapper"
            data-color="primary"
            data-padded={true}
            data-round={0}
            id="example-id"
          >
            <p>
              Content
            </p>
            <div
              className="md-popover-arrow-wrapper"
              id="arrow"
            >
              <Arrow>
                <svg
                  className="md-popover-arrow-svg"
                  data-side="bottom"
                  height={12}
                  viewBox="0 0 24 12"
                  width={24}
                  xmlns="http://www.w3.org/svg/2000"
                >
                  <defs>
                    <clipPath
                      id="popover-arrow-cut-stroke-bottom"
                    >
                      <path
                        d="m 0 12 l 10 -10 q 2 -2 4 0 l 10 10"
                      />
                    </clipPath>
                  </defs>
                  <path
                    clipPath="url(#popover-arrow-cut-stroke-bottom)"
                    d="m 0 12 l 10 -10 q 2 -2 4 0 l 10 10"
                    data-color="primary"
                  />
                </svg>
              </Arrow>
            </div>
          </div>
        </ContentContainer>
      </Portal>
    </Tippy>
  </ForwardRef(TippyWrapper)>
</Popover>
`;

exports[`<Popover /> snapshot should match snapshot with style 1`] = `
<Popover
  style={
    Object {
      "color": "pink",
    }
  }
  triggerComponent={
    <button>
      Click Me!
    </button>
  }
>
  <ForwardRef(TippyWrapper)
    animation={false}
    appendTo="parent"
    offset={
      Array [
        0,
        16,
      ]
    }
    placement="bottom"
    popperOptions={
      Object {
        "modifiers": Array [
          Object {
            "name": "arrow",
            "options": Object {
              "element": "#arrow",
              "padding": 5,
            },
          },
        ],
      }
    }
    render={[Function]}
    trigger="click"
  >
    <Tippy
      animation={false}
      appendTo="parent"
      offset={
        Array [
          0,
          16,
        ]
      }
      placement="bottom"
      popperOptions={
        Object {
          "modifiers": Array [
            Object {
              "name": "arrow",
              "options": Object {
                "element": "#arrow",
                "padding": 5,
              },
            },
          ],
        }
      }
      render={[Function]}
      trigger="click"
    >
      <button>
        Click Me!
      </button>
      <Portal
        containerInfo={
          <div
            data-tippy-root=""
            id="tippy-4"
            style="pointer-events: none; z-index: 9999;"
          >
            <div
              class="md-popover-wrapper"
              data-color="primary"
              data-padded="true"
              data-round="0"
              style="color: pink;"
            >
              <p>
                Content
              </p>
              <div
                class="md-popover-arrow-wrapper"
                id="arrow"
              >
                <svg
                  class="md-popover-arrow-svg"
                  data-side="bottom"
                  height="12"
                  viewBox="0 0 24 12"
                  width="24"
                  xmlns="http://www.w3.org/svg/2000"
                >
                  <defs>
                    <clippath
                      id="popover-arrow-cut-stroke-bottom"
                    >
                      <path
                        d="m 0 12 l 10 -10 q 2 -2 4 0 l 10 10"
                      />
                    </clippath>
                  </defs>
                  <path
                    clip-path="url(#popover-arrow-cut-stroke-bottom)"
                    d="m 0 12 l 10 -10 q 2 -2 4 0 l 10 10"
                    data-color="primary"
                  />
                </svg>
              </div>
            </div>
          </div>
        }
      >
        <ContentContainer
          attrs={
            Object {
              "data-placement": undefined,
            }
          }
          style={
            Object {
              "color": "pink",
            }
          }
        >
          <div
            className="md-popover-wrapper"
            data-color="primary"
            data-padded={true}
            data-round={0}
            style={
              Object {
                "color": "pink",
              }
            }
          >
            <p>
              Content
            </p>
            <div
              className="md-popover-arrow-wrapper"
              id="arrow"
            >
              <Arrow>
                <svg
                  className="md-popover-arrow-svg"
                  data-side="bottom"
                  height={12}
                  viewBox="0 0 24 12"
                  width={24}
                  xmlns="http://www.w3.org/svg/2000"
                >
                  <defs>
                    <clipPath
                      id="popover-arrow-cut-stroke-bottom"
                    >
                      <path
                        d="m 0 12 l 10 -10 q 2 -2 4 0 l 10 10"
                      />
                    </clipPath>
                  </defs>
                  <path
                    clipPath="url(#popover-arrow-cut-stroke-bottom)"
                    d="m 0 12 l 10 -10 q 2 -2 4 0 l 10 10"
                    data-color="primary"
                  />
                </svg>
              </Arrow>
            </div>
          </div>
        </ContentContainer>
      </Portal>
    </Tippy>
  </ForwardRef(TippyWrapper)>
</Popover>
`;
